common/statistics/machine/netStatistics.go

```diff
@@ -22,18 +20,47 @@ type netStatistics struct {
 	totalBytesSentInEpoch     uint64
 	totalBytesReceivedInEpoch uint64
 	getStatisticsHandler      func() ([]net.IOCountersStat, error)
+	cancel                    func()
 }
 
 // NewNetStatistics returns a new instance of the netStatistics
 func NewNetStatistics() *netStatistics {
-	return &netStatistics{
-		getStatisticsHandler: getStatistics,
+	stats := newNetStatistics(getStatistics)
+	stats.startProcessLoop(stats.processLoop)
+
+	return stats
+}
+
+func newNetStatistics(getStatisticsHandler func() ([]net.IOCountersStat, error)) *netStatistics {
+	stats := &netStatistics{
+		getStatisticsHandler: getStatisticsHandler,
+	}
+
+	return stats
+}
+
+func (ns *netStatistics) startProcessLoop(handler func(ctx context.Context)) {
+	var ctx context.Context
+	ctx, ns.cancel = context.WithCancel(context.Background())
+
+	go handler(ctx)
```

Comment:
perhaps create 2 new constructors for tests only that have the handler as parameter? this way `newNetStatistics` won't be needed and this line would become `go ns.stats.processLoop(ctx)`

Reply:
refactored